RW Disaggregate with s3 tiflash crash for Unsupport type: Invalid

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
rw disaggregated with ks3, consistency case
tiflash crash
```
[2023/07/12 14:14:33.136 +08:00] [ERROR] [BaseDaemon.cpp:368] ["(from thread 99) Terminate called after throwing an instance of DB::Exception\nCode: 49, e.displayText() = DB::Exception: Unsupport type: Invalid, e.what() = DB::Exception\nStack trace:\n\n\n       0x759827d\tterminate_handler() [tiflash+123306621]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:633\n  0x7fa27fe9fa13\tstd::__terminate(void (*)()) [libc++abi.so.1+236051]\n  0x7fa27fe9f9b8\tstd::terminate() [libc++abi.so.1+235960]\n       0x1c3672b\t__clang_call_terminate [tiflash+29583147]\n       0x7e6c853\tstd::__1::__packaged_task_func<DB::DM::Remote::DataStoreS3::getDataFilesInfo(std::__1::unordered_set<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&)::$_3,"] [source=BaseDaemon] [thread_id=327]
[2023/07/12 14:14:33.136 +08:00] [ERROR] [BaseDaemon.cpp:376] [########################################] [source=BaseDaemon] [thread_id=327]
[2023/07/12 14:14:33.136 +08:00] [ERROR] [BaseDaemon.cpp:377] ["(from thread 99) Received signal Aborted(6)."] [source=BaseDaemon] [thread_id=327]
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
TiFlash
Release Version: v7.3.0-alpha
Edition:         Community
Git Commit Hash: c6944e7e01ac592fdeb189b490196c967f03a02b
Git Branch:      heads/refs/tags/v7.3.0-alpha
UTC Build Time:  2023-07-11 11:17:12
Enable Features: jemalloc sm4(GmSSL) avx2 avx512 unwind thinlto
Profile:         RELWITHDEBINFO

Raft Proxy
Git Commit Hash:   bc0ebe45a0183c0afb67db609157119776210d5e
Git Commit Branch: HEAD
UTC Build Time:    2023-07-11 11:21:54
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Storage Engine:    tiflash
Prometheus Prefix: tiflash_proxy_
Profile:           release
Enable Features:   Unknown (env var does not exist when building)

--- a/dbms/src/Storages/Page/V3/PageDirectory.cpp
+++ b/dbms/src/Storages/Page/V3/PageDirectory.cpp
@@ -529,8 +529,8 @@ void VersionedPageEntries<Trait>::copyCheckpointInfoFromEdit(const typename Page
     // (the checkpoint info) each page's data was dumped.
     // In this case, there is a living snapshot protecting the data.
 
-    // Pre-check: All ENTRY edit record must contain checkpoint info for copying.
     RUNTIME_CHECK(edit.type == EditRecordType::VAR_ENTRY);
+    // The checkpoint_info from `edit` could be empty when we upload the manifest without any page data
     if (!edit.entry.checkpoint_info.has_value())
         return;
 

--- a/dbms/src/Storages/Page/V3/PageDirectory.cpp
+++ b/dbms/src/Storages/Page/V3/PageDirectory.cpp
@@ -531,7 +531,8 @@ void VersionedPageEntries<Trait>::copyCheckpointInfoFromEdit(const typename Page
 
     // Pre-check: All ENTRY edit record must contain checkpoint info for copying.
     RUNTIME_CHECK(edit.type == EditRecordType::VAR_ENTRY);
-    RUNTIME_CHECK(edit.entry.checkpoint_info.has_value());
+    if (!edit.entry.checkpoint_info.has_value())
+        return;
 
     auto page_lock = acquireLock();
 
@@ -1853,14 +1854,6 @@ size_t PageDirectory<Trait>::copyCheckpointInfoFromEdit(const PageEntriesEdit & 
     const auto & records = edit.getRecords();
     if (records.empty())
         return num_copied;
-
-    // Pre-check: All ENTRY edit record must contain checkpoint info.
-    // We do the pre-check before copying any remote info to avoid partial completion.
-    for (const auto & rec : records)
-    {
-        if (rec.type == EditRecordType::VAR_ENTRY)
-            RUNTIME_CHECK_MSG(rec.entry.checkpoint_info.has_value(), "try to copy checkpoint from an edit with invalid record: {}", rec);
-    }
 
     for (const auto & rec : records)
     {

--- a/dbms/src/Storages/Page/V3/PageEntriesEdit.cpp
+++ b/dbms/src/Storages/Page/V3/PageEntriesEdit.cpp
@@ -64,11 +64,16 @@ typename PageEntriesEdit<UniversalPageId>::EditRecord PageEntriesEdit<UniversalP
     rec.being_ref_count = 1;
     if (rec.type == EditRecordType::VAR_ENTRY)
     {
-        rec.entry.checkpoint_info = OptionalCheckpointInfo{
-            .data_location = CheckpointLocation::fromProto(proto_edit.entry_location(), strings_map),
-            .is_valid = true,
-            .is_local_data_reclaimed = true,
-        };
+        // uploading page data may be disabled
+        auto checkpoint_loc = CheckpointLocation::fromProto(proto_edit.entry_location(), strings_map);
+        if (checkpoint_loc.isValid())
+        {
+            rec.entry.checkpoint_info = OptionalCheckpointInfo{
+                .data_location = std::move(checkpoint_loc),
+                .is_valid = true,
+                .is_local_data_reclaimed = true,
+            };
+        }
         rec.entry.size = proto_edit.entry_size();
         rec.entry.checksum = proto_edit.entry_checksum();
         rec.entry.tag = proto_edit.entry_tag();

--- a/dbms/src/Storages/Page/V3/PageEntryCheckpointInfo.h
+++ b/dbms/src/Storages/Page/V3/PageEntryCheckpointInfo.h
@@ -47,6 +47,8 @@ struct CheckpointLocation
         const CheckpointProto::EntryDataLocation & proto_rec,
         CheckpointProto::StringsInternMap & strings_map);
 
+    bool isValid() const { return !data_file_id->empty(); }
+
     std::string toDebugString() const
     {
         return fmt::format("{{data_file_id: {}, offset_in_file: {}, size_in_file: {}}}", *data_file_id, offset_in_file, size_in_file);

